openhands/server/listen.py

```diff
@@ -452,6 +482,138 @@ async def get_security_analyzers():
+
+    # Get prompt template - for now using default
+    prompt_template = ''
+
+    try:
+        await resolve_github_issue(
+            owner=resolve_request.owner,
+            repo=resolve_request.repo,
+            token=resolve_request.token,
+            username=resolve_request.username,
+            max_iterations=resolve_request.max_iterations,
+            output_dir=output_dir,
+            llm_config=llm_config,
+            runtime_container_image=runtime_container_image,
+            prompt_template=prompt_template,
+            issue_type=resolve_request.issue_type,
+            repo_instruction=None,
+            issue_number=resolve_request.issue_number,
+            comment_id=resolve_request.comment_id,
+            reset_logger=True,
+        )
+
+        # Read output.jsonl file
+        output_file = os.path.join(output_dir, 'output.jsonl')
+        if os.path.exists(output_file):
+            with open(output_file, 'r') as f:
+                return {'status': 'success', 'output': f.read()}
+        else:
+            return {'status': 'error', 'message': 'No output file generated'}
+
+    except Exception as e:
+        return {'status': 'error', 'message': str(e)}
+    finally:
+        # Cleanup temp directory
+        if os.path.exists(output_dir):
+            import shutil
+
+            shutil.rmtree(output_dir)
+
+
+@app.post('/api/resolver/send-pr')
+async def send_pull_request(
+    request: Request, pr_request: SendPullRequestRequest
+) -> dict[str, str | dict[str, Any]]:
+    """Create a pull request or branch for resolved issue.
+
+    This endpoint creates either:
+    - A draft PR
+    - A ready PR
+    - Just a branch
+
+    With the changes made by the resolver.
+
+    Args:
+        request: The incoming request object
+        pr_request: The PR creation request parameters
+
+    Returns:
+        A dictionary containing the PR/branch creation results with keys:
+        - status: 'success' or 'error'
+        - result: PR creation result (on success)
+        - message: Error message (on error)
+    """
+    try:
+        # Load the resolver output for this issue
+        output_dir = os.path.join(tempfile.gettempdir(), 'openhands_resolver')
+        resolver_output = load_single_resolver_output(
+            output_dir, pr_request.issue_number
+        )
+        if not resolver_output:
+            raise ValueError(
+                f'No resolver output found for issue {pr_request.issue_number}'
+            )
+
+        # Get LLM config from current session
+        llm_config = config.get_llm_config()
+
+        # Get GitHub token from environment
+        github_token = os.environ.get('GITHUB_TOKEN')
+        if not github_token:
+            raise ValueError('GITHUB_TOKEN environment variable not set')
+
+        # Get GitHub username from environment
+        github_username = os.environ.get('GITHUB_USERNAME')
+
+        # Process the issue
+        result = await process_single_issue(
```

Comment:
`process_single_issue` doesn't return anything. let's change it to return a pydantic class indicating the result of processing the issue, and modify the code here appropriately.

also, process_single_issue isn't async, so we don't need "await" here.